expan/core/experiment.py

```diff
@@ -287,11 +288,70 @@ def _quantile_filtering(self, data, kpis, percentile, threshold_type):
         :return: boolean values indicating whether the row should be filtered
         :rtype: pd.Series
         """
-        method_table = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
-        na_replacement={'upper': -float_info.max        , 'lower':  float_info.max}
-
+        method_table   = {'upper': lambda x: x > threshold, 'lower': lambda x: x <= threshold}
+        na_replacement = {'upper': -float_info.max,         'lower':  float_info.max}
 
         for column in data[kpis].columns:
             threshold = np.percentile(data[column].fillna(na_replacement[threshold_type]), percentile)
             flags = flags | data[column].apply(method_table[threshold_type])
         return flags
+
+
+    def variant_split_is_consistent_with_expected_distribution(self, variant_column, weights, min_counts=5, alpha=0.05):
+        """ Tests the consistency of variant split with the hypothesized distribution.
+        
+        :param variant_column: variant column from the input data frame
+        :param weights: list of dicts with weights/split percentage per each variant 
+                        ([{'value':<variant>,'weight':<weight>}, ...])
+        :param min_counts: minimum number of observed and expected frequencies (should be at least 5), see 
+                            http://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.chisquare.html
+        :param alpha: significance level, 0.05 by default
+        :return: True(if split is consistent with the given split) or
+                 False(if split is not consistent with the given split)
+        :rtype:  Boolean, float, float
+        """
+        if not variant_column:
+            logger.warning("Variant split check was cancelled since input variant column is empty or doesn't exist.")
+            return None
+        if not weights:
+            logger.warning("Variant split check was cancelled since input weights are empty or doesn't exist.")
+            return None
+        if len(weights) <= 1 or len(variant_column) <= 1:
+            logger.warning("Variant split check was cancelled since input weights or the number if categories "
+                           "is less than 2.")
+            return None
+
+        # Count number of observations per each variant
+        variant_column = pd.Series(variant_column).fillna(0)
```

Comment:
do you need to fill nas ? won't they be ignored anyway later on?

Reply:
or maybe you can drop nas?

Reply:
yes, I don't need it.